gobblin-data-management/src/main/java/gobblin/data/management/copy/replication/AbstractReplicationEndPoint.java

```diff
@@ -0,0 +1,48 @@
+package gobblin.data.management.copy.replication;
+
+import org.testng.collections.Objects;
+
+import gobblin.data.management.copy.CopyableDataset;
+
+
+/**
+ * Abstract class which has the basic implementations 
```

Comment:
This is not a very useful javadoc. Please describe:
- What implementations the class provides
- What implementations are left to the sub-classes.